pkg/streamingpromql/planning.go

```diff
@@ -29,6 +32,14 @@ import (
 	"github.com/grafana/mimir/pkg/util/spanlogger"
 )
 
+var smoothedIncompatibleFunctionPrefix string
+var anchoredIncompatibleFunctionPrefix string
+
+func init() {
+	smoothedIncompatibleFunctionPrefix = fmt.Sprintf("smoothed modifier can only be used with: %s - not with ", sortImplode(promql.SmoothedSafeFunctions))
```

Comment:
Why are these in an `init` block? If there are timing or ordering concerns I would much rather refactor this to have some static part and eat any runtime cost since the error case should be infrequent.